discovery-provider/plugins/notifications/src/processNotifications/mappers/base.ts

```diff
@@ -12,6 +14,73 @@ export abstract class BaseNotification<Type> {
     this.identityDB = identityDB
   }
 
+  async fetchEntities(entityIds: number[], entityType: string) {
+    switch (entityType) {
+      case EntityType.Track: {
+        const res: Array<{ track_id: number; title: string }> = await this.dnDB
+          .select('track_id', 'title')
+          .from<TrackRow>('tracks')
+          .where('is_current', true)
+          .whereIn(
+            'track_id',
+            entityIds.map((id) => id.toString())
+          )
+        const tracks = res.reduce((acc, track) => {
+          acc[track.track_id] = { title: track.title }
+          return acc
+        }, {} as Record<number, { title: string }>)
+        return tracks
+      }
+      case EntityType.Playlist: {
+        const res: Array<{
+          playlist_id: number
+          playlist_name: string
+          is_album: boolean
+        }> = await this.dnDB
+          .select('playlist_id', 'playlist_name', 'is_album')
+          .from<PlaylistRow>('playlists')
+          .where('is_current', true)
+          .whereIn(
+            'playlist_id',
+            entityIds.map((id) => id.toString())
+          )
+        const playlists = res.reduce((acc, playlist) => {
+          acc[playlist.playlist_id] = {
+            playlist_name: playlist.playlist_name,
+            is_album: playlist.is_album
+          }
+          return acc
+        }, {} as Record<number, { playlist_name: string; is_album: boolean }>)
+        return playlists
+      }
+      default:
+        console.error(`Fetching entity type ${entityType} not supported`)
+    }
+  }
+
+  async getUsersBasicInfo(userIds: Array<number>) {
+    const res: Array<{
+      user_id: number
+      name: string
+      is_deactivated: boolean
+    }> = await this.dnDB
+      .select('user_id', 'name', 'is_deactivated')
+      .from<UserRow>('users')
+      .where('is_current', true)
+      .whereIn(
+        'user_id',
+        userIds.map((id) => id.toString())
+      )
+    const users = res.reduce((acc, user) => {
+      acc[user.user_id] = {
+        name: user.name,
+        isDeactivated: user.is_deactivated
```

Comment:
I'm curious about these reducers in each of the query functions. All but this line are copying properties from one place to another individually, which feels tedious and/or error prone. And if I understand correctly, the intent is to return dictionaries of each type keyed by ids.

So two questions:
* Why does this one field use camelCase when the rest are snake case?
* Could we not just use something like lodash `keyBy('id')` or even a simple reduce that spreads the whole object:
```ts
type UserBasicInfo = {
  user_id: number
  name: string
  is_deactivated: boolean
}
//...
return res.reduce<Record<number, UserBasicInfo>>((acc, user) => ({
  ...acc,
  [user.user_id]: {...user}
}), {})
```

Reply:
all great points! lol tbh i have no idea how this ended up this way, i copied existing code and moved it to this file. but i like your suggestions and will use thank you!